### PR TITLE
Solr 8 / Updating deps

### PR DESCRIPTION
**Pull request by AceHack**
Please approve this PR moving to Solr 8
Also update test collection config from solr-8.6.0 release
(directory server/solr/configsets/sample_techproducts_configs/conf)

### CODE REVIEW

**Comment by AceHack** [comment]
@magro

**Comment by magro** [comment]
Hey @AceHack, at first thanks for the PR! Maybe you have missed that, the build is failing with compile errors: https://travis-ci.org/github/inoio/solrs/jobs/707497621

Btw, I had also started to work on Solr 8 some time ago, but due to some reason (I don't remember) did not complete / merge this: https://github.com/inoio/solrs/pull/76

**Comment by AceHack** [comment]
Finally, all tests pass, that was hard lol :)

File: build.sbt [hunk 1]
```diff
@@ -28,24 +28,24 @@ scalacOptions ++= Seq(
...
-initialize := {
-  val _ = initialize.value
-  if (sys.props("java.specification.version") != "1.8")
-    sys.error(s"Java 8 is required for this project. Running: ${sys.props("java.specification.version")}")
-}
+// initialize := {
+//   val _ = initialize.value
+//   if (sys.props("java.specification.version") != "1.8")
+//     sys.error(s"Java 8 is required for this project. Running: ${sys.props("java.specification.version")}")
+// }
```
Comment (magro) [inline build.sbt:35]: (Why) is this change needed?

**Comment by AceHack** [comment]
@magro this was a mistake, I was building locally using Java 11.  I added it back.  Thanks.

**Comment by magro** [comment]
@mkr do you also want to have a look, because I know you also wanted to work on this?

**Comment by mkr** [comment]
@AceHack Thanks, this looks good and comes in handy :)
* It's great that you updated the configs we use in tests. I think we should strip them down a bit and remove some unused stuff to get rid of the many log messages about unresolved libs during tests (i.e. remove the `<lib dir` section and maybe the unused LTR and clustering components config). Or just move to a more minimal config set (such as https://github.com/apache/lucene-solr/tree/master/solr/example/files/conf).
* The Solr 8.6.1 bugfix release is imminent. I think it makes sense to just jump to that version if we have a couple of days.

@magro Is that JDK8 build requirement still required?

**Comment by magro** [comment]
> Is that JDK8 build requirement still required?

When this was added java 8 was the latest version. We still need Java 8+, so this check could be improved to support also newer versions than jdk 8.
Assuming that today nobody would work with an older jdk it's also fine for me to just remove the check completely.

**Comment by AceHack** [comment]
@magro Any chance on getting this merged soon?

**Comment by magro** [comment]
Cool, thanks for the collaboration @AceHack and @mkr !

**Comment by magro** [comment]
Is released as 2.5.0, on it's way to maven central...
